src/View/Helper/FormHelper.php

```diff
@@ -1502,13 +1502,20 @@ protected function _magicOptions($fieldName, $options, $allowOverride)
             unset($options['step']);
         }
 
-        $autoLength = !array_key_exists('maxlength', $options)
-            && !empty($fieldDef['length'])
-            && $options['type'] !== 'select';
-
-        $allowedTypes = ['text', 'textarea', 'email', 'tel', 'url', 'search'];
-        if ($autoLength && in_array($options['type'], $allowedTypes)) {
-            $options['maxlength'] = min($fieldDef['length'], 100000);
+        $typesWithMaxLength = ['text', 'textarea', 'email', 'tel', 'url', 'search'];
+        if (in_array($options['type'], $typesWithMaxLength)) {
+            $validationLength = $context->getMaxLength($fieldName);
```

Comment:
Because this method isn't in the interface you should do a `method_exists()` check.

Reply:
This name could be `$maxLength` as it won't always be based on a validator.

Reply:
thanx updated.